### ISSUE TITLE
Error: Error populating bind file: wrapped C/C++ object of type TextCtrl has been deleted

### ISSUE THREAD

**Issue by autobotcoop** [emersonrp/bindcontrol#54]
![image](https://github.com/emersonrp/bindcontrol/assets/39393574/4b2cd204-0725-4ba5-ab5b-c47241c5d703)
[Stub.zip](https://github.com/user-attachments/files/15796643/Stub.zip)

**Comment by emersonrp** [comment]
That's an interesting one.  Had you deleted any custom binds before that?  Or steps inside custom binds?  I need to work out a recipe to reproduce this.  It's actually a harmless error as far as writing the bind files, but I should totally find it and fix it anyway, as it's indicative of some kind of sloppiness somewhere.

**Comment by autobotcoop** [comment]
Yes I had. 

I also finished testing this by deleting all my binds in a test profile. The error remained.

**Comment by emersonrp** [comment]
OK interesting.  And if you save the Profile, exit the app, and then reload the Profile, does it still do it?  I suspect not, I think it's just a transient state it gets into post-deletion while the app is still open.

If it's not writing a lot of binds, then yeah I'm wrong and I need to get into this sooner than later.  And I might be wrong about the transient state, too.

**Comment by emersonrp** [comment]
If you can pin down a specific recipe, "open this attached profile, delete the bind called 'fubble', write binds," that will help a lot.  Meanwhile I'm trying various things based on it complaining about a TextCtrl, but there are a lot of those to test.

**Comment by emersonrp** [comment]
This is with the "Stub" profile you attached?  I'm not getting that happening with it.  Hmmn.  This happens on "Write Binds" yes?  Or am I jumping to that conclusion?

**Comment by autobotcoop** [comment]
Yes. 

Let me clear the entire directory, and unzip again.

**Comment by autobotcoop** [comment]
Okay, sorry to bother you! 

Looks like there was a fragment from old version causing issues. Wiping the BC directory (`C:\Games\Modding\City of Heroes\BindControl`) and starting completely over fixed it.

**Comment by emersonrp** [comment]
lol I'm actually going to reopen this because I think I've found a way to make it happen.  Load Stub, Write Binds, Load Stub, Write Binds, POW.  The second load is where the badness happens.  Please take a second and see if you can make this happen the same way.

**Comment by autobotcoop** [comment]
Oh my gosh, you're right. 

Why does wiping the directory fix it temporarily then?

**Comment by emersonrp** [comment]
Not sure.  Maybe you didn't load Stub twice when you tested after wiping the directory?  Not clear, but I have a recipe that works to cause a bug so I am on it.

**Comment by emersonrp** [comment]
OK that is fixed.  Yeah if a Profile had been loaded that had Custom Binds in it, then another Profile was loaded, references to the Custom Binds from the first Profile lingered when we tried to build / populate the Custom Binds page with the second Profile.  Those references pointed nowhere, hence it complaining about deleted controls.  Fixed.

I have a vague suspicion there might be some other similar buglets hiding in there.  The Custom Binds page is complicated, lol.

https://github.com/emersonrp/bindcontrol/releases/tag/v0.17.12.4
